### ISSUE TITLE
Timed out while waiting for the machine to boot.

### ISSUE THREAD

**Issue by geoHeil** [dcos/dcos-vagrant#40]
Just trying DCOS for the first time. But Vagrant fails with a timeout.
I am running on a mac-book-pro 16GB RAM

```
==> a1: Checking if box 'mesosphere/dcos-centos-virtualbox' is up to date...
==> a1: Setting the name of the VM: a1.dcos
==> a1: Fixed port collision for 22 => 2222. Now on port 2200.
==> a1: Clearing any previously set network interfaces...
==> a1: Preparing network interfaces based on configuration...
    a1: Adapter 1: nat
    a1: Adapter 2: hostonly
==> a1: Forwarding ports...
    a1: 22 (guest) => 2200 (host) (adapter 1)
==> a1: Running 'pre-boot' VM customizations...
==> a1: Booting VM...
==> a1: Waiting for machine to boot. This may take a few minutes...
    a1: SSH address: 127.0.0.1:2200
    a1: SSH username: vagrant
    a1: SSH auth method: private key
Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
the configured ("config.vm.boot_timeout" value) time period.
```

**Comment by karlkfi** [comment]
Did you follow step 3 (Configure VirtualBox Networking) in the setup?

Does this happen repeatedly if you destroy your cluster (`vagrant destroy -f`) and recreate?

**Comment by activatedgeek** [comment]
Hey @karlkfi, I did use the same technique as in the documentation. Have a look at my `vboxnet0` interface.
```
vboxnet0: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	ether 0a:00:27:00:00:00 
	inet 192.168.65.1 netmask 0xffffff00 broadcast 192.168.65.255
```

I have tried this multiple times and failed.

Here is the complete log for command `vagrant up m1 a1 p1 boot`:
```
Bringing machine 'm1' up with 'virtualbox' provider...
Bringing machine 'a1' up with 'virtualbox' provider...
Bringing machine 'p1' up with 'virtualbox' provider...
Bringing machine 'boot' up with 'virtualbox' provider...
==> m1: Importing base box 'mesosphere/dcos-centos-virtualbox'...
==> m1: Matching MAC address for NAT networking...
==> m1: Checking if box 'mesosphere/dcos-centos-virtualbox' is up to date...
==> m1: Setting the name of the VM: m1.dcos
==> m1: Clearing any previously set network interfaces...
==> m1: Preparing network interfaces based on configuration...
    m1: Adapter 1: nat
    m1: Adapter 2: hostonly
==> m1: Forwarding ports...
    m1: 22 (guest) => 2222 (host) (adapter 1)
==> m1: Running 'pre-boot' VM customizations...
==> m1: Booting VM...
==> m1: Waiting for machine to boot. This may take a few minutes...
    m1: SSH address: 127.0.0.1:2222
    m1: SSH username: vagrant
    m1: SSH auth method: private key
    m1: Warning: Remote connection disconnect. Retrying...
    m1: Warning: Connection timeout. Retrying...
    m1: Warning: Connection timeout. Retrying...
Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
the configured ("config.vm.boot_timeout" value) time period.

If you look above, you should be able to see the error(s) that
Vagrant had when attempting to connect to the machine. These errors
are usually good hints as to what may be wrong.

If you're using a custom box, make sure that networking is properly
working and you're able to connect to the machine. It is a common
problem that networking isn't setup properly in these boxes.
Verify that authentication configurations are also setup properly,
as well.

If the box appears to be booting properly, you may want to increase
the timeout ("config.vm.boot_timeout") value.
```

Here is the `VagrantConfig.yaml`:
```yaml
m1:
  ip: 192.168.65.90
  cpus: 1
  memory: 1024
  type: master
a1:
  ip: 192.168.65.111
  cpus: 1
  memory: 3072
  type: agent-private
p1:
  ip: 192.168.65.60
  cpus: 1
  memory: 1024
  type: agent-public
  aliases:
  - spring.acme.org
  - oinker.acme.org
boot:
  ip: 192.168.65.50
  cpus: 1
  memory: 256
  type: boot
```

**Comment by geoHeil** [comment]
Yes indeed I set the networking.

**Comment by ploddi** [comment]
Same issue. First run cluster works just fine, but after `vagrant halt` or `vagrant suspend` it not starting correctly.

**Comment by karlkfi** [comment]
@ploddi: halt and suspend have not been tested and not explicitly supported by DC/OS itself. Agent nodes can come and go, but if the master zookeeper loses quorum, it's a manual process to restore. Easier to destroy and recreate the cluster.

@activatedgeek & @geoHeil: There are some suggestions to try here: http://stackoverflow.com/questions/23293071/timed-out-while-waiting-for-the-machine-to-boot-when-vagrant-up failure to boot would be a problem with the base os image, not a problem with DC/OS itself, which hasn't been installed at that point.

**Comment by pdxmholmes** [comment]
Same issue as the others, same basic log. When I have the VirtualBox GUI open you can clearly see what the issue is:

http://i.imgur.com/xgtfzjk.png

It's stuck at a login prompt. The only weird thing I saw was that when I "pre-added" the Vagrant box, it downloaded 0.5.0. When I run 'vagrant up' it wants to use version 0.4.2 of that image.

**Comment by depoorterp** [comment]
Also facing this problem on Windows 10 x64 VBox 5.0.18.
Created the host network, but its not called vboxnet0 on windows, but something like "VirtualBox Host-Only Ethernet Adapter #2", which is referenced by the VBox definition for e.g. m1 in the network settings. I do get some additional information over the others.

The m1 is marked as up after vagrant status and I can vagrant ssh in the machine.

```
    ...  
    m1: Warning: Remote connection disconnect. Retrying...
    m1:
    m1: Vagrant insecure key detected. Vagrant will automatically replace
    m1: this with a newly generated keypair for better security.
Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
the configured ("config.vm.boot_timeout" value) time period.

If you look above, you should be able to see the error(s) that
Vagrant had when attempting to connect to the machine. These errors
are usually good hints as to what may be wrong.

If you're using a custom box, make sure that networking is properly
working and you're able to connect to the machine. It is a common
problem that networking isn't setup properly in these boxes.
Verify that authentication configurations are also setup properly,
as well.

If the box appears to be booting properly, you may want to increase
the timeout ("config.vm.boot_timeout") value.
```

**Comment by karlkfi** [comment]
Don't have a lot of experience running Vagrant on Windows. I'm not sure if this is a box issue or a setup issue.

Can all of you list your OS, OS version, Vagrant version, and VirtualBox version?

**Comment by mrchris2000** [comment]
Same as depoorterp above: Windows 10 x64 VBox 5.0.18, latest Vagrant version (1.8.1).
Same issue as well.

**Comment by saquibmian** [comment]
I'm also running into the same issue; same configs as @activatedgeek 

OS X Yosemite, 10.11.4 (15E65), Vagrant 1.8.1, VirtualBox 5.0.18-106667

**Comment by geoHeil** [comment]
Same for me osx 10.11.4, vagrant 1.8.1 Virtual Box 5.0.16 r105871

**Comment by karlkfi** [comment]
@saquibmian You mean OSX El Capitan 10.11, right? Yosemite is 10.10 (which I have locally).

I can get an El Capitan machine to test on. I know there were significant changes to security lock down in El Capitan. That may or may not be part of the problem, given the reports of issues on Windows 10 too.

**Comment by saquibmian** [comment]
@karlkfi yes, you're right -- I'm running El Capitan. 
I don't think System Integrity Protection would interfere with Vagrant much, but I do recall there being issues with the VirtualBox kernel extension when I upgraded to EC. That was "resolved" in a later version of VirtualBox. 
It would be good to test against El Capitan though.

**Comment by karlkfi** [comment]
I just tested successfully first time on a machine with a fresh install of OS X El Capitan 10.11.4 (16gb ram).

I didn't have to install a new version of git or ruby, just used what came with the latest developer tools. Because I didn't have to use Homebrew, I didn't have to bypass the System Integrity Protection.

```
$ git --version
git version 2.6.4 (Apple Git-63)
$ ruby --version
ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin15]
$ vagrant --version
Vagrant 1.8.1
$ VBoxManage --version
5.0.18r106667
```

So I get the feeling y'all have some non-standard issues that I don't know how to reproduce or debug. I would suggest trying other vagrant boxes and seeing if you have the same problem.

**Comment by seyhbold** [comment]
Same here on Windows 7 Pro, Virtualbox 5.0.18 and Vagrant 1.8.1 with embedded ruby. Other boxes are working.

**Comment by karlkfi** [comment]
Can you guys run this with `export VAGRANT_LOG=debug` or `--debug`? Share the output in a gist, please. If it's a login problem the log should say `all authorization methods failed (tried none, publickey, password)` or similar.

If it's a publickey problem, then that would seem to indicate that your Vagrant's public/private key pair is corrupt or modified somehow. The base OS image box includes the default vagrant "insecure" public key (https://github.com/mitchellh/vagrant/tree/master/keys) to facilitate first time login, but it also is configured with the default "insecure" vagrant credentials as well (vgrant/vagrant). Login should fall back to user/pass if SSH key auth fails, but it's possible it's only trying one method or not trying the right method.

**Comment by saquibmian** [comment]
@karlkfi here you go: https://gist.github.com/saquibmian/b09eab3a99e71350037b487725c2af53

There's over 40,000 lines there!

My vagrant insecure key appears to be in working order -- I'm able to boot/ssh into public vagrant images deployed locally.

**Comment by karlkfi** [comment]
@vishnu2kmohan has pointed me at some networking workarounds for cleaning up network interfaces and disabling Network Manager that may resolve the issue, but will require building a new OS image:
https://bugzilla.redhat.com/show_bug.cgi?id=1221006
https://github.com/chef/bento/blob/master/scripts/centos/cleanup.sh
https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux_OpenStack_Platform/3/html/Installation_and_Configuration_Guide/Disabling_Network_Manager.html

So I made some changes to the box builder:
https://github.com/dcos/dcos-vagrant-box/commit/adf0f00395a9ca96625697e705b328e76c1dbded

I'll build a dev box tomorrow and y'all can try it to see if it helps. In the meantime you can try a local build, if you want.
Build: https://github.com/dcos/dcos-vagrant-box#manual-build
Test: https://github.com/dcos/dcos-vagrant-box#test-local-box

**Comment by sahil-sharma** [comment]
Greetings,
I am also facing the same timeout error while spinning up (m1 a1 p1 boot) and also not able to access (http://m1.dcos). Though able to ssh into (m1 a1 p1). Have created vboxnet0 and assigned IP (192.168.65.1/24) but not a single machine (m1 a1 p1) got IP from that subnet.
How can I access my dcos? Any suggestions would be of great help!

Here are my environment details:
Vagrant 1.8.1
OS: Ubuntu 14.04/16.04 (64-bit) 4.4.0-21-generic
RAM: 16GB, i5
Virtualbox Version 5.0.18r106667

**Comment by karlkfi** [comment]
```

**Comment by karlkfi** [comment]
Beta box is uploaded. Use the following to test it:

```
wget https://raw.githubusercontent.com/dcos/dcos-vagrant-box/dcos-centos-virtualbox-0.5.1/metadata.json
export DCOS_BOX_URL=metadata.json
export DCOS_BOX_VERSION=0.5.1
```

note: you have to download the metadata file first for this method or else vagrant gets confused and thinks it's a corrupted box file.

To revert back to the official boxes:

```
unset DCOS_BOX_URL
unset DCOS_BOX_VERSION
vagrant box remove mesosphere/dcos-centos-virtualbox --box-version=0.5.1
echo -n "https://downloads.dcos.io/dcos-vagrant/metadata.json" > ~/.vagrant.d/boxes/mesosphere-VAGRANTSLASH-dcos-centos-virtualbox/metadata_url
```

**Comment by karlkfi** [comment]
Another option is to upgrade virtualbox, which has had some recent NAT bug fixes: https://www.virtualbox.org/wiki/Changelog

**Comment by sahil-sharma** [comment]
I updated my virtualbox to 5.0.20 but still facing the same timeout error while executing the vagrant up m1 command. I tried the beta box too and then reverted back to previous box. Tried multiple times with fresh start. But failed every time.

**Comment by depoorterp** [comment]
Didn't succeed with the new image either. I did augment the boot_timeout to 6 minutes and seem to get furher in the installation. Got The VM Booted message after which I can ssh into the vm. But the vm is not very responsive as long as the vagrant up is running.

I'm going to follow-up on some things i picked-up on the internet that pottentially can speed up things on the windows/vagrant/vbox combination.

**Comment by webmutation** [comment]
@karlkfi Has anyone successfully run this on a Windows machine? I just tested on a fresh copy of Windows 10 and Windows 7 with no luck, on my home PC. 

Unfortunately the organization that I work, has windows on all 1200+ Developer PCs and we have zero permission to run any other OS. Im trying to run this on a 24Gb developer desktop at the institution I work.

**Comment by AntoineGa** [comment]
i succeeded by doing up / halt for each VM, as halting VM seems  to successfully update ssh keys. So if i want to up m1 a1 a2 p1 boot, i had to up m1, halt m1, up m1 then, up a1, halt a1, up a1, and so on till boot.

**Comment by surdy** [comment]
@webmutation  @AntoineGa I have had some partial success. Working on making it work consistently. I'll update soon and hopefully send a PR

**Comment by surdy** [comment]
I had to modify the dcos-vagrant box to make it work https://github.com/dcos/dcos-vagrant-box/pull/4 (In Review)

**Comment by karlkfi** [comment]
heh, auto closed by merge...

Can those having trouble with this symptom (windows especially) try the following new dev release:

```
wget https://downloads.dcos.io/dcos-vagrant/dcos-centos-virtualbox-0.5.2.box
vagrant box add dcos-centos-virtualbox-0.5.2.box --name mesosphere/dcos-centos-virtualbox
export DCOS_BOX_VERSION=0
vagrant up m1 a1 p1 boot
```

**Comment by seyhbold** [comment]
Great I got past the ssh key stuff with this.

I am on Windows 7 Professional
Virtualbox 5.0.18
vagrant 1.8.1
dcos on tag v0.6.0

For the first test I just did a `vagrant up boot` to see if the machine starts.
Now I am going to try to bring up all.

**Comment by webmutation** [comment]
Tried the new setup same issue persists.

> > vagrant up m1 a1 p1 boot
> Bringing machine 'm1' up with 'virtualbox' provider...
> Bringing machine 'a1' up with 'virtualbox' provider...
> Bringing machine 'p1' up with 'virtualbox' provider...
> Bringing machine 'boot' up with 'virtualbox' provider...
> ==> m1: Checking if box 'mesosphere/dcos-centos-virtualbox' is up to date...
> ==> m1: Clearing any previously set forwarded ports...
> ==> m1: Clearing any previously set network interfaces...
> ==> m1: Preparing network interfaces based on configuration...
>     m1: Adapter 1: nat
>     m1: Adapter 2: hostonly
> ==> m1: Forwarding ports...
>     m1: 22 (guest) => 2222 (host) (adapter 1)
> ==> m1: Running 'pre-boot' VM customizations...
> ==> m1: Booting VM...
> ==> m1: Waiting for machine to boot. This may take a few minutes...
>     m1: SSH address: 127.0.0.1:2222
>     m1: SSH username: vagrant
>     m1: SSH auth method: private key
>     m1: Warning: Remote connection disconnect. Retrying...
>     m1: Warning: Remote connection disconnect. Retrying...
>     m1: Warning: Remote connection disconnect. Retrying...
>     m1:
>     m1: Vagrant insecure key detected. Vagrant will automatically replace
>     m1: this with a newly generated keypair for better security.
> ==> m1: Machine booted and ready!
> ==> m1: Checking for guest additions in VM...
> ==> m1: Setting hostname...
> An error occurred in the underlying SSH library that Vagrant uses.
> The error message is shown below. In many cases, errors from this
> library are caused by ssh-agent issues. Try disabling your SSH
> agent or removing some keys and try again.
> 
> If the problem persists, please report a bug to the net-ssh project.
> 
> timeout during server version negotiating

**Comment by karlkfi** [comment]
@webmutation `timeout during server version negotiating` sounds like a different issue than the `Timed out while waiting for the machine to boot.` error other people were experiencing.

Can you make a [DC/OS JIRA](https://dcosjira.atlassian.net/) ticket to report it so we can track it separately? If you make the ticket with an account, you'll get notifications of updates.

Preliminary research indicates that the ssh timeout is hard coded to 15s in vagrant: https://github.com/mitchellh/vagrant/blob/master/plugins/communicators/ssh/communicator.rb#L342

PR to increase to 30s got rejected: https://github.com/mitchellh/vagrant/pull/7014

It's possible that some other configuration tweak might decrease the ssh connection time, tho.

**Comment by surdy** [comment]
@webmutation the issue you encountered is a different issue as @karlkfi mentioned. I think it might be related to https://github.com/mitchellh/vagrant/issues/6812. Some people did report that installing vbguest plugin solved the problem for them : `vagrant plugin install vagrant-vbguest`

**Comment by karlkfi** [comment]
I made a new dev box with updated vbox guest additions, if you want to try that: https://downloads.dcos.io/dcos-vagrant/dcos-centos-virtualbox-0.6.0.box

**Comment by sahil-sharma** [comment]
Greetings DC/OS Team,

I tried to set-up DC/OS cluster (m1 a1 p1 boot) with a fresh start and downloaded new vagrant box [mesosphere/dcos-centos-virtualbox (virtualbox, 0.6.0)].
With a partial success I am able to spin-up (m1 a1 p1 boot) in one go but only at first time. If I try for second time then I have to spin-up one at a time.
After I got my cluster up and running I tried to access http://m1.dcos/ from my web browser and I tried to register myself with my Google account but unable to login.
![dcos gui error](https://cloud.githubusercontent.com/assets/5774379/15283199/3ef00814-1b66-11e6-846c-316454476c95.png)

I tried to visit this link providing this link after 3 unsuccessful attempts but got 404 error.
[](https://dcos.io/docs/1.7/administration/security-and-authentication/)
Kindly see and do the needful.
Thank you.

**Comment by sahil-sharma** [comment]
Finally I got the dashboard. I ran the browser with root user root@machine1: # firefox &
Then I ran http://m1.dcos and got the dashboard. :)

**Comment by activatedgeek** [comment]
@sahil-sharma Great that you got it working! Can you confirm the result of a command for me? On any of your cluster nodes, can you run `nslookup google.com` and tell me if you get positive results. (You can use any equivalent for `nslookup`, I just want results for a DNS resolution query). Please run with both `root` and `non-root` user.

**Comment by karlkfi** [comment]
@sahil-sharma That symptom is being tracked here: https://github.com/dcos/dcos-vagrant/issues/58

**Comment by sahil-sharma** [comment]
@activatedgeek By default there is no dig/nslookup command in m1 so was failed to get the desired results. But when I installed it using (yum install -y bind-utils) as DC/OS have used CentOS then I got the desired results as (both root and non-root user): 
[root@m1 vagrant]# nslookup google.com
Server:		198.51.100.1
Address:	198.51.100.1#53
Non-authoritative answer:
Name:	google.com
Address: 216.58.197.78

[vagrant@m1 ~]$ nslookup google.com
Server:		198.51.100.1
Address:	198.51.100.1#53
Non-authoritative answer:
Name:	google.com
Address: 216.58.197.46

**Comment by sahil-sharma** [comment]
As I have posted in my earlier post that I was not able to login via using my github/google account but after entering login credentials if we refresh the page multiple time then we can get the dashboard despite of having an error message.
Secondly, I tried to install spark and for that I executed 3 nodes (a1 a2 a3) and assigned mentioned resources (2gb RAM and 2 CPU cores) but still not able to install spark either from Universe section from DC/OS dashboard or using dcos cli (./bin/dcos package install spark). Any inputs?
Did I follow the correct approach or missing out something? Kindly suggest me.
Thanks.
